hcloud/error.go

```diff
@@ -100,6 +101,15 @@ type Error struct {
 }
 
 func (e Error) Error() string {
+	if resp := e.Response(); resp != nil {
+		isServerError := resp.StatusCode >= http.StatusInternalServerError && resp.StatusCode <= 599
```

Comment:
```suggestion
		isServerError := resp.StatusCode >= 500 && resp.StatusCode <= 599
```

Maybe we should not mix the constants and the digits?